src/pep8ext_naming.py

```diff
@@ -55,10 +55,18 @@ def err(self, node, code: str, **kwargs):
 
 
 class NameSet(frozenset[str]):
-    """A set of names that are matched using fnmatchcase."""
-
-    def __contains__(self, item) -> bool:
-        return any(fnmatchcase(item, name) for name in self)
+    """A set of names that can be matched as Unix shell-style wildcards."""
+    _fnmatch: bool = False
+
+    def __new__(cls, iterable: Iterable[str]):
+        obj = super().__new__(cls, iterable)
+        obj._fnmatch = any(c in r"*?[" for name in iterable for c in name)
```

Comment:
Is it faster to loop over the characters in the name or the 3 characters in the raw string which doesn't strictly need to be raw? 

Also would it be faster still to turn a name into a frozen set and look for the intersection of that with the frozen set of those three characters?

Reply:
It's actually fastest (by 2-3x) to check one character at a time, presumable because we hit a `memchr` or similar fast path internally:

```py
any("*" in s or "?" in s or "[" in s for s in iterable)
```